integration-tests/deployment/multiclient.go

```diff
@@ -95,30 +99,72 @@ func (mc *MultiClient) CodeAt(ctx context.Context, account common.Address, block
 
 func (mc *MultiClient) NonceAt(ctx context.Context, account common.Address) (uint64, error) {
 	var count uint64
-	err := mc.retryWithBackups(func(client *ethclient.Client) error {
+	err := mc.retryWithBackups("NonceAt", func(client *ethclient.Client) error {
 		var err error
 		count, err = client.NonceAt(ctx, account, nil)
 		return err
 	})
 	return count, err
 }
 
-func (mc *MultiClient) retryWithBackups(op func(*ethclient.Client) error) error {
+func (mc *MultiClient) WaitMined(ctx context.Context, tx *types.Transaction) (*types.Receipt, error) {
```

Comment:
@AnieeG @connorwstein here is a parallel wait mined that i used in the Confirm of `chainlink-deployments`. working well.